### PR TITLE
Convert PicassoExecutorService.java to PicassoExecutorService.kt

### PR DESCRIPTION
**Pull request by amilcar-andrade**

### CODE REVIEW

File: picasso/src/main/java/com/squareup/picasso3/PicassoExecutorService.kt [hunk 1]
```diff
@@ -0,0 +1,57 @@
...
+/**
+ * The default {@link java.util.concurrent.ExecutorService} used for new {@link Picasso} instances.
+ * <p>
+ * Exists as a custom type so that we can differentiate the use of defaults versus a user-supplied
+ * instance.
+ */
+internal class PicassoExecutorService(threadFactory: ThreadFactory) : ThreadPoolExecutor(
+    DEFAULT_THREAD_COUNT, DEFAULT_THREAD_COUNT, 0, TimeUnit.MILLISECONDS,
+    PriorityBlockingQueue(), threadFactory
+) {
```
Comment (jrodbx) [inline picasso/src/main/java/com/squareup/picasso3/PicassoExecutorService.kt:34]:
```suggestion
internal class PicassoExecutorService(
  threadFactory: ThreadFactory
) : ThreadPoolExecutor(
    DEFAULT_THREAD_COUNT, DEFAULT_THREAD_COUNT, 0, TimeUnit.MILLISECONDS,
    PriorityBlockingQueue(), threadFactory
) {
```

File: picasso/src/main/java/com/squareup/picasso3/PicassoExecutorService.kt [hunk 2]
```diff
@@ -0,0 +1,57 @@
...
+import java.util.concurrent.Future
+import java.util.concurrent.FutureTask
+import java.util.concurrent.PriorityBlockingQueue
+import java.util.concurrent.ThreadFactory
+import java.util.concurrent.ThreadPoolExecutor
+import java.util.concurrent.TimeUnit
+
+/**
+ * The default {@link java.util.concurrent.ExecutorService} used for new {@link Picasso} instances.
+ * <p>
```
Comment (jrodbx) [inline picasso/src/main/java/com/squareup/picasso3/PicassoExecutorService.kt:27]:
```suggestion
 *
```

File: picasso/src/main/java/com/squareup/picasso3/PicassoExecutorService.kt [hunk 3]
```diff
@@ -0,0 +1,57 @@
...
+) {
+
+    override fun submit(task: Runnable): Future<*> {
+        val futureTask = PicassoFutureTask(task as BitmapHunter)
+        execute(futureTask)
+        return futureTask
+    }
+
+    private class PicassoFutureTask(private val hunter: BitmapHunter) : FutureTask<BitmapHunter>(
+            hunter, null), Comparable<PicassoFutureTask> {
```
Comment (jrodbx) [inline picasso/src/main/java/com/squareup/picasso3/PicassoExecutorService.kt:43]:
```suggestion
    private class PicassoFutureTask(
      private val hunter: BitmapHunter
    ) : FutureTask<BitmapHunter>(hunter, null), Comparable<PicassoFutureTask> {
```

File: picasso/src/main/java/com/squareup/picasso3/PicassoExecutorService.kt [hunk 4]
```diff
@@ -0,0 +1,57 @@
...
+
+            // High-priority requests are "lesser" so they are sorted to the front.
+            // Equal priorities are sorted by sequence number to provide FIFO ordering.
+            return if (p1 == p2) hunter.sequence - other.hunter.sequence else p2.ordinal - p1.ordinal
+        }
+    }
+
+    companion object {
+        private const val DEFAULT_THREAD_COUNT = 3
+    }
```
Comment (jrodbx) [inline picasso/src/main/java/com/squareup/picasso3/PicassoExecutorService.kt:56]: nit: general formatting.  eventually we'll add a linter, but for now, 2 space indentation would be ideal.

**Review by jrodbx — APPROVED** [review]

**Comment by amilcar-andrade** [comment]
@jrodbx ooops, I was going to make the changes but seems got merged by accident. Thanks for reviewing tho

**Pull request closed without merging**